screeninfo/screeninfo.py

```diff
@@ -1,398 +1,22 @@
-            ('x', ctypes.c_uint32), ('y', ctypes.c_uint32),
-            ('width', ctypes.c_uint32), ('height', ctypes.c_uint32),
-            ('mode_valid', ctypes.c_int),
-            ('mode', DrmModeModeInfo),
-            ('gamma_size', ctypes.c_int),
-        ]
-
-        def __del__(self):
-            if self.need_free:
-                libdrm.drmModeFreeCrtc(ctypes.byref(self))
-
-    libdrm = load_library('drm')
-    libdrm.drmModeGetResources.argtypes = [ctypes.c_int]
-    libdrm.drmModeGetResources.restype = ctypes.POINTER(DrmModeRes)
-    libdrm.drmModeFreeResources.argtypes = [ctypes.POINTER(DrmModeRes)]
-    libdrm.drmModeFreeResources.restype = None
-    libdrm.drmModeGetConnector.argtypes = [ctypes.c_int, ctypes.c_uint32]
-    libdrm.drmModeGetConnector.restype = ctypes.POINTER(DrmModeConnector)
-    libdrm.drmModeFreeConnector.argtypes = [ctypes.POINTER(DrmModeConnector)]
-    libdrm.drmModeFreeConnector.restype = None
-    libdrm.drmModeGetEncoder.argtypes = [ctypes.c_int, ctypes.c_uint32]
-    libdrm.drmModeGetEncoder.restype = ctypes.POINTER(DrmModeEncoder)
-    libdrm.drmModeFreeEncoder.argtypes = [ctypes.POINTER(DrmModeEncoder)]
-    libdrm.drmModeFreeEncoder.restype = None
-    libdrm.drmModeGetCrtc.argtypes = [ctypes.c_int, ctypes.c_uint32]
-    libdrm.drmModeGetCrtc.restype = ctypes.POINTER(DrmModeCrtc)
-    libdrm.drmModeFreeCrtc.argtypes = [ctypes.POINTER(DrmModeCrtc)]
-    libdrm.drmModeFreeCrtc.restype = None
-
-    monitors = []
-
-    for card_no in range(DRM_MAX_MINOR):
-        card_path = DRM_DEV_NAME % (DRM_DIR_NAME, card_no)
-        try:
-            fd = os.open(card_path, os.O_RDONLY)
-        except OSError:
-            continue
-        if fd < 0:
-            continue
-
-        res = libdrm.drmModeGetResources(fd).contents
-        res.fd = fd
-        res.needs_free = True
-        for connector in res.connectors:
-            if connector.connection == DrmModeConnector.DRM_MODE_CONNECTED:
-                crtc = connector.encoder.crtc
-                monitors.append(
-                    Monitor(crtc.x, crtc.y, crtc.width, crtc.height))
-        os.close(fd)
-
-    return monitors
-
-
-def _enumerate_osx():
-    from pyobjus import autoclass
-    from pyobjus.dylib_manager import load_framework, INCLUDE
-    load_framework(INCLUDE.AppKit)
-
-    screens = autoclass('NSScreen').screens()
-    monitors = []
-
-    for i in range(screens.count()):
-        f = screens.objectAtIndex_(i).frame
-        if callable(f):
-            f = f()
-
-        monitors.append(
-            Monitor(f.origin.x, f.origin.y, f.size.width, f.size.height))
-
-    return monitors
-
+"""
+Author: Marcin Kurczewski and others
+License: MIT License
+Copyright (C) 2015 Marcin Kurczewski
+"""
 from sys import platform
+from ._cygwin import enumerate_cygwin
+from ._drm import enumerate_drm
+from ._osx import enumerate_osx
+from ._windows import enumerate_windows
+from ._x11 import enumerate_x11
+from ._utils import load_library
```

Comment:
What do PEPs and coding practices have to say about naming modules with leading undescores? I have never used that pattern before. I think it's a bit silly and would name them normally but throw them in a submodule such as `.detail`.

Reply:
It looks a bit strange, yes, but I chose to use this approach because attributes starting with an underscore are protected, and this can extend to files in packages. These files are not for use outside the package, and thus the filename starts with an underscore. If you prefer, I will remove the underscores from the filenames.